silencer-plugin/src/main/scala/com/github/ghik/silencer/FilterType.scala

```diff
@@ -0,0 +1,43 @@
+package com.github.ghik.silencer
+
+import scala.util.matching.Regex
+import com.github.ghik.silencer.FilterType.{RegexFilter, StringFilter}
+
+sealed trait FilterType {
```

Comment:
To be quite honest, I don't see much value in this abstraction layer. I'd rather keep things simple and have three plain fields in `SuppressingReporter`: `globalFilters: List[Regex]`, `sourceRoots: List[File]` and `pathFilters: List[Regex]`.

Reply:
allows for less code duplication, I can remove it. I didn't want to pass 3 extra fields to the constructor, and wanted to give room for further expansion.

Also, the logic in the `SilencerPlugin` now allows for passing the same parameter multiple times, without the need to necessarily separate the values via `;`, for example:

`-Psilencer:globalFilters=depreFunc1\\ in\\ object\\ globallyFiltered\\ is\\ deprecated -Psilencer:globalFilters=useless.*filter`

which does make for slightly more readable/understandable complex patterns

Reply:
It may be worth to aggregate all the options into a single class in order to avoid passing too many parameters to `SilencerPlugin`. But I think each option should still be handled independently and literally on its own. They don't really have much in common. Yes, `globalFilters` and `pathFilters` are both of the same type (`List[Regex]`) but this is rather accidental. It should be enough to just have a common `private def` for parsing these two. No reason to introduce abstractions for *all* possible options.